extensions/investspec/investspec.py

```diff
@@ -0,0 +1,358 @@
+    elif arg_type == 'csv':
+        return f'`CSV <{INPUT_TYPES_HTML_FILE}#csv>`__'
+    else:
+        return f'`{arg_type} <{INPUT_TYPES_HTML_FILE}#{arg_type}>`__'
+
+
+def format_required_string(required):
+    """Represent an arg's required status as a user-friendly string.
+
+    Args:
+        required (bool | str | None): required property of an arg. May be
+            `True`, `False`, `None`, or a conditional string.
+
+    Returns:
+        string
+    """
+    if required is None or required is True:
+        return 'required'
+    elif required is False:
+        return 'optional'
+    else:
+        # assume that the about text will describe the conditional
+        return 'conditionally required'
+
+
+def format_geometries_string(geometries):
+    """Represent a set of allowed vector geometries as user-friendly text.
+
+    Args:
+        geometries (set(str)): set of geometry names
+
+    Returns:
+        string
+    """
+    # sort the geometries so they always display in a consistent order
+    sorted_geoms = sorted(
+        geometries,
+        key=lambda g: GEOMETRY_ORDER.index(g))
+    return ', '.join(geom.lower() for geom in sorted_geoms)
+
+
+def format_permissions_string(permissions):
+    """Represent a rwx-style permissions string as user-friendly text.
+
+    Args:
+        permissions (str): rwx-style permissions string
+
+    Returns:
+        string
+    """
+    permissions_strings = []
+    if 'r' in permissions:
+        permissions_strings.append('read')
+    if 'w' in permissions:
+        permissions_strings.append('write')
+    if 'x' in permissions:
+        permissions_strings.append('execute')
+    return ', '.join(permissions_strings)
+
+
+def format_options_string_from_dict(options):
+    """Represent a dictionary of option: description pairs as a bulleted list.
+
+    Args:
+        options (dict): the dictionary of options to document, where keys are
+            options and values are descriptions of the options
+
+    Returns:
+        list of RST-formatted strings, where each is a line in a bullet list
+    """
+    lines = []
+    # casefold() is a more aggressive version of lower() that may work better
+    # for some languages to remove all case distinctions
+    sorted_options = sorted(
+        list(options.keys()),
+        key=lambda option: option.casefold()
+    )
+    for option in sorted_options:
+        lines.append(f'- {option}: {options[option]}')
+    return lines
+
+
+def format_options_string_from_list(options):
+    """Represent options as a comma-separated list.
+
+    Args:
+        options (list[str]): the set of options to document
+
+    Returns:
+        string of comma-separated options
+    """
+    return ', '.join(options)
+
+
+def capitalize(name):
+
+    def capitalize_word(word):
+        if word in {'of', 'the'}:
+            return word
+        else:
+            return word[0].upper() + word[1:]
```

Comment:
unfortunately `.capitalize()` and `.title()` also lowercase all the other letters:
```
>>> 'REDD land cover map'.capitalize()
'Redd land cover map'
>>> 'REDD land cover map'.title()
'Redd Land Cover Map'
```

Reply:
The `REDD` case is an interesting one because it's a bit more subtle ... I hadn't considered an acronym at all, and you're right!  Could you add a comment to this to clarify why what you have here is the correct solution?